lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -210,6 +210,15 @@ def parse_scope_name(scope_name: str, vo: Optional[str]) -> tuple[str, ...]:
     :raises ValueError: when scope_name could not be parsed.
     :returns: a (scope, name) tuple.
     """
+
+    encoded_slash = r"%2F"
+    if encoded_slash in scope_name:
+        if scope_name.count('/') != 1:
```

Comment:
I'm a bit confused - how can there be a `/` in the `scope_name` if this is the path where slashes in `scope_name` are encoded?

Reply:
It is confusing.

The request sent looks something like `/did/scope/name%2Fwith%2Fslash%2F`.

Since we are using the `AllowEncodedSlashes NoDecode` then this will be preserved. If we were not using this, then the `scope_name` variable will be converted into: `scope/name/with/slash`. It would be ambiguous the distinction between scope and name and also the trailing slash is lost, so we cannot differentiate between did with and without trailing slash for some requests.

Reply:
Thanks, I see what you mean now. It might be worth adding a comment to clarify this just to make it a bit simpler to read through the code - e.g. something that explains that the single `/` that we expect to see is the one that divides scope and name

Reply:
Added a comment to explain.